src/Datadog.Trace.ClrProfiler.Managed.Core/TypeExtensions.cs

```diff
@@ -51,6 +51,37 @@ internal static class TypeExtensions
             return null;
         }
 
+        public static System.Type GetInstrumentedType(
+            this object runtimeObject,
+            string instrumentedTypeName,
+            string[] instrumentedNamespaces)
+        {
+            if (runtimeObject == null)
+            {
+                return null;
+            }
+
+            var currentType = runtimeObject.GetType();
+
+            while (currentType != null)
+            {
+                if (currentType.Name == instrumentedTypeName)
+                {
+                    foreach (var instrumentedNamespace in instrumentedNamespaces)
+                    {
+                        if (currentType.Namespace == instrumentedNamespace)
```

Comment:
Thinking out loud for a later PR, I wonder if there's a faster way we can do this comparison 🤔

Reply:
I didn't need this new overload anymore so I removed it in c1d01a27.